R/tm_landing_popup.R

```diff
@@ -0,0 +1,82 @@
+#' Landing Popup Module
+#'
+#' @description This module creates a landing welcome popup for `teal` applications.
+#'
+#' @details If you use this module in `teal::init(modules = )`, it will not be wrapped in a tab in `teal` application.
+#' See `teal::module(type = 'teal_module_landing')`.
+#'
+#' @param label `character(1)` the label of the module.
+#' @param title `character(1)` the text to be displayed as a title of the popup.
+#' @param content `character(1)` the content of the popup. Passed to `...` of `shiny::modalDialog`. Can be a `character`
+#' or a text input control (like `textInput`) or a list of `shiny` tags. See examples.
+#' @param buttons `shiny` tag or a list of tags (`tagList`). Typically a `modalButton` or `actionButton`. See examples.
+#'
+#' @return A `teal_module_landing` module to be used in `teal` applications.
+#'
+#' @examples
+#' app1 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal.modules.general::tm_landing_popup(
+#'       content = "A place for the welcome message or a disclaimer statement.",
+#'       buttons = modalButton("Proceed")
+#'     ),
+#'     teal.modules.general::tm_front_page()
+#'   )
+#' )
+#' if (interactive()) {
+#'   shinyApp(app1$ui, app1$server)
+#' }
+#'
+#' app2 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal.modules.general::tm_landing_popup(
+#'       title = "Welcome",
+#'       content = div(tags$b("A place for the welcome message or a disclaimer statement.", style = "color: red;")),
+#'       buttons = tagList(
+#'         modalButton("Proceed"),
+#'         actionButton("close", "Read more", onclick = "window.open('http://google.com', '_blank')")
```

Comment:
Perhaps create an example with a "Reject" button that will close the app? Navigate to a new tab in 5 seconds?

Reply:
Yup, just extended. Good idea
https://github.com/insightsengineering/teal/pull/934/commits/4eb1a39995b910efb69aac5a853479fb92e55ace